pkgs/applications/networking/remote/tlclient/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, copyDesktopItems
+, makeDesktopItem
+
+, alsaLib
+, libX11
+, pcsclite
+}:
+let
+  tlclient-png = fetchurl {
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/tlclient.png?h=tlclient&id=fd126edd54a012caa8c9ce47ab8d7fb25e982e74";
+      sha256 = "12i0wc4a8r69kwcdj0kjidwvmwhpb8jcwsqxw9dl5vm1gdqxyz5v";
+      name = "tlclient-png";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "tlclient";
+  version = "4.12.0-6517";
+
+  src = fetchurl {
+    # x86_64 arch tarball
+    url = "https://www.cendio.com/downloads/clients/tl-${version}-client-linux-dynamic-x86_64.tar.gz";
+    sha256 = "0qpn28sh3p13f581gmdl3j59m40iwlpm6bpar7qiz1kr7cwlrzim";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook copyDesktopItems ];
+  propagatedBuildInputs = [ alsaLib libX11 pcsclite ];
+
+  dontBuild = true;
+
+  installPhase = ''
+     install -Dm644 "lib/tlclient/EULA.txt" "$out/share/licenses/tlclient/EULA.txt"
+     install -m644 "lib/tlclient/open_source_licenses.txt" "$out/share/licenses/tlclient/open_source_licenses.txt"
+     cp -R lib "$out/"
+
+    install -Dm644 "etc/tlclient.conf" "$out/etc/tlclient.conf"
+    install -Dm755 "bin/tlclient" "$out/bin/tlclient"
+    install -Dm755 "bin/tlclient-openconf" "$out/bin/tlclient-openconf"
+
+    install -Dm644 "${tlclient-png}" "$out/share/tlclient/tlclient.png"
+  '';
```

Comment:
It is missing the desktop item, I think because you forgot to add the `runHook preInstall` at the start of `installPhase` and `runHook postInstall` at the end of it.

Reply:
Without it, `copyDesktopItems` hook will not run since it runs at `postInstall` phase (I think).

Reply:
Fixed.